src/crimson/os/seastore/extent_placement_manager.cc

```diff
@@ -167,7 +167,10 @@ SegmentedAllocator::Writer::write(
                   allocated_to);
                 auto rsize = record.get_encoded_record_length();
                 return (num_extents ?
-                        _write(t, record, rsize) :
+                        [this, &record, &rsize, &t] {
+                          allocated_to += rsize.mdlength + rsize.dlength;
+                          return _write(t, record, rsize);
```

Comment:
nit, move shared logic (probably getting rsize and modify allocated_to) into `_write()`?

Reply:
ok, I'll do that once we resolve this [conversation](https://github.com/ceph/ceph/pull/43483#discussion_r726744809), thanks:-)